File: backend/app/core/upgrades/curses/damage_reduc_1.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class DamageReduc1(UpgradeImplementation):
    id = 'damage_reduc_1'
    name = 'Weaknes 1'
    description = '-0.1% damage.'
    rarity = 'common'
    type = 'stat_modifier'
    cost = 0

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = -0.001
        appliers.modifiers.add_multiplier('attack', 'damage', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True

File: frontend/src/game/upgrades/curses/damage_reduc_1.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const DamageReduc1Def: UpgradeDef = {
  id: "damage_reduc_1",
  name: "Weakness 1",
  description: "-0.1% damage.",
  rarity: RarityID.Common,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 0,
  targetClass: UpgradeTargetID.Attack,
  fieldInTargetClass: UpgradeStatID.Damage,
  value: -0.001,
  isMultiplier: true,
  stackable: true,
  maxStacks: 99999,
  curse: true,
}

export class DamageReduc1 extends Upgrade {}
